### DISCUSSION TITLE
Make the Vehicle struct accepts more than 32 peeps

### DISCUSSION THREAD

**Discussion by frutiemax** [OpenRCT2/OpenRCT2#19521] [Feature requests]
Internally, the vehicle structure is limited to 32 peeps. This is a blocking issue for rides that would require more than 32 peeps, unless you use multiple vehicles. This would be a good refactor to permit 255 peeps.

**Comment by ZehMatt** [comment]
What kind of ride do you have in mind that holds 255 peeps? Need to be a bit careful when it comes to increasing structures.

**Comment by spacek531** [comment]
I have been imagining show rides that can hold peeps in a grandstand, that could reasonably hold a lot of peeps, but I think vehicles would have a hard time reaching 64 guests much less 255. 64 is a stretch for moving vehicles because of the limitations caused by OpenRCT2's strip-drawing method. RCT is also extremely geared toward peeps being paired up, so realistically the upper limit would be 254.

**Reply by ocalhoun6** [reply]
Don't see any reason to limit it in the code to 254, though ... since there are *some* rides in RCT already that accept an odd number of guests per vehicle.

Though I kind of doubt anyone would ever want to make a (realistic) ride that maxes this out anyway. The upper limits would probably only see use as something like a hacked transport ride that's not meant to be realistic, but just meant to be an efficient (and kind of cheating) way to move huge numbers of guests at a time. Even just getting 255 guests through the queue line and loaded into a vehicle would take *ages*, and that's if there's only one vehicle. If you hacked a monorail to have 255 guests per vehicle in a long train of vehicles, I could see problems with having guests thinking "I want to get off of X" before the train is even finished loading. And anywhere such a train *unloads* is instantly going to have severe overcrowding problems.

So ... yeah. I don't think people will actually be trying to push the limit very much. Just that they might want to have certain rides with 40 or 50 guests per vehicle.

**Reply by spacek531** [reply]
The game can only handle peeps in pairs or individually. 255 individual peep sprites for just one single angle of the vehicle... yikes.

**Reply by ocalhoun6** [reply]
Not all vehicles have the peep sprites visible, such as the monorail.

**Reply by spacek531** [reply]
Is there a ride people want that would have >64 peeps and doesn't have sprites? Like yeah ok you could make a 6x6 cinema ride but what's the point?

**Reply by ocalhoun6** [reply]
Hacked monorails, used to transport massive numbers of guests?

**Comment by frutiemax** [comment]
I was starting coding a giant ferris wheel but I got stuck due to the vehicle limitations. Then recently I've went with multiple vehicles approach for the condor ride so this is a non-issue for this. However, this required some custom code for the vehicle placement, similar to the dodgems.

I've looked at the code and it does not seem that simple to increase the number off peeps per ride...

**Comment by frutiemax** [comment]
https://www.sixflags.com/larondeen/attractions/titan

This one would be hard to do with multiple vehicles (40 passengers). Perhaps could shrink it to 32 passengers since rct peeps are not exactly the same form factor as humans...

**Reply by ocalhoun6** [reply]
It would also be useful for making a [Disk'O ride](https://d302e0npexowb4.cloudfront.net/wp-content/uploads/2016/10/52cfba65aa21a-disko-coaster-02.jpg).

Edit: another ride with more than 32 passengers per vehicle: https://www.ride-extravaganza.com/assets/rainbow_riders.jpg
